consolidate.py

```diff
@@ -0,0 +1,145 @@
+    website: str
+    authors: List[Author]
+    docs: Optional[str]
+    readme: Optional[str]
+    company: Optional[Company]
+    support: Optional[str]
+
+
+@dataclasses.dataclass
+class RepositoryEntry:
+    identifier: str
+    name: str
+    description: str
+    docker: str
+    versions: Dict[str, Version]
+    extension_logo: Optional[str]
+    company_logo: Optional[str]
+
+
+class Consolidator:
+    registry = Registry()
+    consolidated_data: List[Dict[str, Any]] = []
+
+    async def fetch_readme(self, url):
+        if not url.startswith("http"):
+            print(f"This is not a valid url: '{url}'; Assuming it is an inline readme")
+            return url
+        async with aiohttp.ClientSession() as session:
+            async with session.get(url) as resp:
+                if resp.status != 200:
+                    print(f"Error status {resp.status}")
+                    raise Exception(f"Could not get readme {url}")
+                return await resp.text()
+
+    async def all_repositories(self):
+        repos = Path("./repos")
+        for repo in repos.glob("**/*.json"):
+            with open(repo, "r", encoding="utf-8") as individual_file:
+                try:
+                    data = json.load(individual_file)
+                except Exception as exc:
+                    raise Exception(f"Unable to parse file {repo}") from exc
+                extension_logo = (repo / "../extension_logo.png").resolve().relative_to(repos.resolve())
+                company_logo = (repo / "../company_logo.png").resolve().relative_to(repos.resolve())
+                readme = data.get("readme", None)
+                if readme:
+                    readme = await self.fetch_readme(readme)
+                try:
+                    new_repo = RepositoryEntry(
+                        identifier=data["identifier"],
+                        name=data["name"],
+                        docker=data["docker"],
+                        description=data["description"],
+                        extension_logo=f"{REPO_ROOT}/{extension_logo}" if extension_logo else "",
+                        versions={},
+                        company_logo=f"{REPO_ROOT}/{company_logo}" if company_logo else None,
+                    )
+                    yield new_repo
+                except Exception as error:
+                    raise Exception(f"unable to read file {individual_file}: {error}") from error
+
+    async def run(self):
+        async for repository in self.all_repositories():
+            for tag in await self.registry.fetch_remote_tags(repository.docker):
+                try:
+                    raw_labels = await self.registry.fetch_labels(f"{repository.docker}:{tag}")
+                    permissions = raw_labels.get("permissions", None)
+                    website = raw_labels.get("website", None)
+                    authors = raw_labels.get("authors", None)
+                    docs = raw_labels.get("docs", None)
```

Comment:
This seems at least marginally annoying. Should `self.registry.fetch_labels` potentially return a `collections.defaultdict` instead, with the default factory set to `lambda: None`? Opinionated, so fine to leave as is if you don't agree / if it's annoying to change.